### PR TITLE
MDS-3354: Fix Name Issue With Party Organizations

### PR DESCRIPTION
**Pull request by exzizt**
# Main
* Fixes an issue where parties of type "organization" could be created with a "first name" field, causing data quality issues
* Sets `first_name` to null for all existing parties that are an organization
# Other
* Fixes an issue where a party's "name" hybrid property and expression was not being correctly calculated

### CODE REVIEW

File: services/core-api/app/api/parties/party/models/party.py [hunk 1]
```diff
@@ -1,55 +1,60 @@
...
+
 
 class Party(SoftDeleteMixin, AuditMixin, Base):
     __tablename__ = 'party'
+
     party_guid = db.Column(UUID(as_uuid=True), primary_key=True, server_default=FetchedValue())
-    first_name = db.Column(db.String, nullable=True)
-    middle_name = db.Column(db.String, nullable=True)
+    first_name = db.Column(db.String)
+    # TODO: Determine if this column is necessary; it appears to not be used by anything.
```
Comment (exzizt) [inline services/core-api/app/api/parties/party/models/party.py:23]: Can we delete this column? It's not being used anywhere.

File: services/core-api/app/api/parties/party/models/party.py [hunk 2]
```diff
@@ -100,15 +102,20 @@ def json(self, show_mgr=True, relationships=[]):
             context.update({
                 'mine_party_appt': [item.json() for item in self.mine_party_appt],
             })
+
         return context
 
+    @name.expression
```
Comment (exzizt) [inline services/core-api/app/api/parties/party/models/party.py:108]: Funnily enough, their example uses a "name" and I copied its structure for the `expression`.
https://docs.sqlalchemy.org/en/13/orm/mapped_sql_expr.html#using-a-hybrid

File: services/core-api/app/api/parties/party/models/party.py (hunk 1 above)
Comment (mmumby) [reply services/core-api/app/api/parties/party/models/party.py:23]: you want to delete "first_name"  or "middle_name"?

"middle_name" yes, I'd delete - not sure why we have that.

**Review by mmumby — DISMISSED** [review]
looks good, Luke!

**Review by mmumby — DISMISSED** [review]

**Review by mmumby — APPROVED** [review]

**Pull request merged**